pgspecial/dbcommands.py

```diff
@@ -103,17 +105,17 @@ def list_roles(cur, pattern, verbose):
                 u.valuntil as rolvaliduntil,
                 ARRAY(SELECT g.groname FROM pg_catalog.pg_group g WHERE u.usesysid = ANY(g.grolist)) as memberof
             FROM pg_catalog.pg_user u
-            """
+            """)
 
     params = {}
     if pattern:
         _, schema = sql_name_pattern(pattern)
-        sql += "WHERE r.rolname ~ %(rolname)s"
-        params["rolname"] = schema
-    sql += " ORDER BY 1"
+        sql += SQL("WHERE r.rolname ~ %(rolname)s")
```

Comment:
Same here, I may be mistaken, but from [the docs](https://www.psycopg.org/psycopg3/docs/api/sql.html#) looks like this should now be:

`sql += SQL("WHERE r.rolname ~ {rolname}")`

vs 

`sql += SQL("WHERE r.rolname ~ %(rolname)s")`

Reply:
If you use the `{name}` parameters, you have to use `.format()` to pass it a value, which must be a Composible instance. If it's not, it will be wrapped in a Literal (in psycopg) and adapted as per normal rules (in psycopg2 it will raise an exception, FYI).

If you use a `%s` or `%(name)s` placeholder, it will not be touched by format, but then you will pass the values to replace on execute.

So, the full picture should be one of the two:

```python
cur.execute(sql.SQL("WHERE rolename = {rolename}").format(rolename=myrolename))
cur.execute(sql.SQL("WHERE rolename = %(rolename)s"), {"rolename": myrolename})
```